data_spec_v1.md

```diff
@@ -0,0 +1,164 @@
+# Initial specification
+
+Provisional, as discussed 5/7/17
+
+## Contents
+- [Intro](#intro)
+- [Notes](#notes)
+- [User Roles](#user-roles)
+- [API Permissions](#api-permissions)
+- [Data Model](#data-model)
+
+## Intro
+The specification for the Nazareth open tourism platform v0.
+
+In essence it is an open data platform where anyone can add and view data on (initially) users, places, events, products.
+
+There will be a permissions system where, in simple terms, moderators and SUPER users will have various edit permissions for the data.
+
+## Notes
+Things not addressed in this file yet:
+- multi-language support
+    - we think we will append/prepend db fields with 'en' or 'ar'
+- Http logging
+    - we think this will be just a simple db table logging request info
+- login and signup flow
+    - e.g. email verification?
+
+## User Roles
+Unless otherwise specified, the below roles have any permissions assigned to roles lower in the list below.
+
+- SUPER - Owner of the open platform.
+- ADMIN - moderator can create any data -> can assign ownership of content
+- OWNER - Specific to resource.
+- BASIC - can have ownership of organisations / their data / events / products
+
+## Data Model
+
+### Users
+
+#### CRUD Permissions
+
+name | permissions
+---|---
+CREATE | role=BASIC:ADMIN, role=ADMIN:SUPER
+READ | isPublic ? any : OWNER
+UPDATE | only OWNER or SUPER
+DELETE | on OWNER or SUPER
+
+#### Fields
+
+
+name | type | required | permissions
+---|---|---|---
+id | KEY | true |-
+createdAt| DATE | - | -
+updatedAt| DATE | - | -
+isPublic | BOOL | true | only OWNER or SUPER (not ADMIN?)
+username (do we need this?) | TEXT | false | only OWNER or SUPER
```

Comment:
I guess there's a fairly likely future feature that will be something along the lines of reviews or a CMS type thing. In which case we will want authors for the content, i.e. usernames? Or that could be done manually with each post or whatever.

Using mongo if we start off without and just log in using email and password, adding usernames if needed in the future shouldn't be too hard?